packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2571,10 +2577,30 @@ class _RouteEntry extends RouteTransitionRecord {
   }
 }
 
+enum _ObservedAction {
```

Comment:
As an idea: maybe use a polymorphic approach: 

* remove this enum
* have an abstract `void notify(List<Observers> observers)` on _NavigatorObservation
* have subclasses for the various actions that implement the correct notification dispatch for the observation.

That also avoids the big switch below